include/boost/geometry/extensions/gis/projections/impl/pj_init.hpp

```diff
@@ -204,8 +205,14 @@ inline parameters pj_init(R const& arguments, bool use_defaults = true)
 
     if (! s.empty())
     {
-        // TODO: IMPLEMENT SPLIT
-        pin.to_meter = atof(s.c_str());
+        std::size_t const divisor = s.find('/');
+        if (divisor == std::string::npos)
+            pin.to_meter = std::atof(s.c_str());
+        else
+        {
+            std::string const numerator(s.substr(0, divisor)), denominator(s.substr(divisor + 1));
+            pin.to_meter = std::atof(numerator.c_str()) / std::atof(denominator.c_str());
```

Comment:
Sure, that of course can be done easily, but does it really make sense
given that the input is text from the source code? I mean it's not like the
usual run-time case where the user enters a bad string and can enter a good
one next time: if this fails it is because the string is wrong in the code,
right?
On 09/10/2015 8:08 AM, "Barend Gehrels" <notifications@github.com> wrote:

> In include/boost/geometry/extensions/gis/projections/impl/pj_init.hpp
> <https://github.com/boostorg/geometry/pull/324#discussion_r41569150>:
>
> > @@ -204,8 +205,14 @@ inline parameters pj_init(R const& arguments, bool use_defaults = true)
> >
> >      if (! s.empty())
> >      {
> > -        // TODO: IMPLEMENT SPLIT
> > -        pin.to_meter = atof(s.c_str());
> > +        std::size_t const divisor = s.find('/');
> > +        if (divisor == std::string::npos)
> > +            pin.to_meter = std::atof(s.c_str());
> > +        else
> > +        {
> > +            std::string const numerator(s.substr(0, divisor)), denominator(s.substr(divisor + 1));
> > +            pin.to_meter = std::atof(numerator.c_str()) / std::atof(denominator.c_str());
>
> Please check for denominator=0
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/boostorg/geometry/pull/324/files#r41569150>.
>